From 4d106cc5d8fd328d39b1db3c57572dd3dec915b5 Mon Sep 17 00:00:00 2001
From: Nicholas Clark <nick@ccl4.org>
Date: Thu, 22 Aug 2013 11:48:29 +0200
Subject: [PATCH] For SDBM_File, stop EU::MM from generating its default
 subdirs rule.

The default subdirs rule creates a race condition with the rule that
Makefile.PL explicitly adds to generate libsdbm.a, which can cause parallel
makes to fail.
---
 ext/SDBM_File/Makefile.PL |  6 ++++++
 pod/perldelta.pod         | 11 ++++++++++-
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git ext/SDBM_File/Makefile.PL ext/SDBM_File/Makefile.PL
index fa16a79..b96b80b 100644
--- ext/SDBM_File/Makefile.PL
+++ ext/SDBM_File/Makefile.PL
@@ -23,6 +23,12 @@ WriteMakefile(
 	      PERL_MALLOC_OK => 1,
              );
 
+# We don't want the default subdir rule, as it creates a race condition with the
+# rule we add below.
+sub MY::subdir_x {
+    return '';
+}
+
 sub MY::postamble {
   if ($^O =~ /MSWin32/ && !defined($ENV{SYSTEMROOT})) {
 	if ($Config{'make'} =~ /dmake/i) {
-- 
1.8.2.1

